paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -313,6 +341,75 @@ internal class IntentConfirmationHandler(
         )
     }
 
+    private fun launchGooglePay(
+        googlePay: PaymentConfirmationOption.GooglePay,
+        arguments: Args,
+    ) {
+        if (googlePay.config.merchantCurrencyCode == null && !arguments.initializationMode.isProcessingPayment) {
+            val message = "GooglePayConfig.currencyCode is required in order to use " +
+                "Google Pay when processing a Setup Intent"
+
+            logger.warning(message)
+
+            onIntentResult(
+                Result.Failed(
+                    cause = IllegalStateException(message),
+                    message = R.string.stripe_something_went_wrong.resolvableString,
+                    type = ErrorType.Internal,
+                )
+            )
```

Comment:
can the merchant see the error in this case?

Reply:
The merchant will have be able to see the error in the ADB logs but that it's mainly. We should have a better way of exposing this error for the merchant in test mode. Something we can think about outside of this PR (this just imports what `PaymentSheet` was already doing).

Reply:
Maybe we could use [UserFacingLogger](https://github.com/stripe/stripe-android/blob/72df0dcd97818e693c353fadc9a5b76545f44f86/stripe-core/src/main/java/com/stripe/android/core/utils/UserFacingLogger.kt#L4) for this? They don't even see the message from the `logger` here. But not sure if they would see the failure from the intent result. Agreed doing this outside of this PR makes sense though